lib/plugins/rest.js

```diff
@@ -22,20 +34,61 @@ exports.register = function (server, options, next) {
 		path    : "/v1/instances",
 
 		handler : function (request, reply) {
+			var result;
 			var ami  = request.payload.ami;
 			var type = request.payload.type;
 			instances.createInstance(ami, type)
 			.then(function (instance) {
+				result = instance;
+				return awsAdapter.runInstances(instance);
+			})
+			.then(function (instance) {
 				reply(instance).created("/v1/instances/" + instance.id);
 			})
 			.catch(function (error) {
+				var failedInstance;
 				if (error.name === "ValidationError") {
 					reply("Bad Request: Please check the parameters passed.").code(400);
 				}
+				else if (error.name === "InvalidAMIID.Malformed") {
+					failedInstance = new Instance({
+						id    : result.id,
+						type  : result.type,
+						ami   : result.ami,
+						state : "failed",
+						uri   : null
+					});
+
+					instances.updateInstance(failedInstance);
+					reply("The AMI entered does not exist. Ensure it is of the form ami-xxxxxx.").code(400);
+				}
+				else if (error.name === "InvalidParameterValue") {
+					failedInstance = new Instance({
+						id    : result.id,
+						type  : result.type,
+						ami   : result.ami,
+						state : "failed",
+						uri   : null
+					});
+
+					instances.updateInstance(failedInstance);
+					reply("The Type entered does not exist. Ensure it is a valid EC2 type.").code(400);
+				}
+				else if (error.name === "AuthFailure") {
```

Comment:
Would a `switch` statement be a better choice for this branching structure since they all just check `error.name`?